src/components/pages/home.jsx

```diff
@@ -11,35 +11,37 @@ import config from '../../config'
 import SearchBarComponent from './searchbar'
 export const searchInputId = 'search-input'
 
-const HomePage = React.createClass({
-  contextTypes: {
+class HomePage extends React.Component {
+  static contextTypes = {
     router: routerShape.isRequired,
     searchQuery: PropTypes.string.isRequired,
     searchResults: PropTypes.array.isRequired,
     setSearchQuery: PropTypes.func.isRequired,
-  },
-  propTypes: {
+  };
+
+  static propTypes = {
     countryPackageName: PropTypes.string.isRequired,
     countryPackageVersion: PropTypes.string.isRequired,
     intl: intlShape,
     location: locationShape.isRequired,
     parameters: PropTypes.objectOf(parameterShape).isRequired,
     variables: PropTypes.objectOf(variableShape).isRequired,
-  },
+  };
+
+  state = {inputValue: ''};
```

Comment:
`this.state`?

Reply:
That wouldn't work:

Here, we are defining the initial value of a class attribute. We are not within a method, and using `this` would trigger a syntax error. 

We would use `this` if we initialized the state from a constructor, which is [another way](https://stackoverflow.com/questions/46714799/how-to-set-initial-state-with-using-es6-class-in-react).

This changeset actually have been done by an official React [migration tool](https://github.com/reactjs/react-codemod).